assets/lokomotive-kubernetes/aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl

```diff
@@ -161,6 +161,12 @@ storage:
             --net=host \
             --dns=host \
             --exec=/bootkube -- start --asset-dir=/assets "$@"
+    - path: /etc/tmpfiles.d/etcd-wrapper.conf
```

Comment:
Hmm, rather than having it decoupled from etcd, why can't we use systemd drop-in and use `install` to ensure that directory exists before the service is started?

It seems to me, that this is a slight abuse of  `/etc/tmpfiles.d`.

Reply:
This change will go away once https://github.com/flatcar-linux/coreos-overlay/pull/524 is available in stable.

Reply:
> why can't we use systemd drop-in and use `install` to ensure that directory exists before the service is started?

The dir is already there and created by `systemd-tmpfile` service using the permissions and information as provided in https://github.com/flatcar-linux/coreos-overlay/blob/main/app-admin/etcd-wrapper/files/etcd-wrapper.conf.

> It seems to me, that this is a slight abuse of  `/etc/tmpfiles.d`.

Why is that can you elaborate?

Reply:
>Why is that can you elaborate?

From [documentation](https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html): `tmpfiles.d — Configuration for creation, deletion and cleaning of volatile and temporary files`

The directory we create is neither volatile or temporary. The primary usecase for `tmpfiles.d` is to create the directories/files in `tmpfs` directories, which are wiped after a reboot.

#

>The dir is already there and created by systemd-tmpfile service using the permissions and information as provided in https://github.com/flatcar-linux/coreos-overlay/blob/main/app-admin/etcd-wrapper/files/etcd-wrapper.conf.

Ah, I see, hmm... Given that we planned not to rely on Flatcar's `etcd-wrapper`, IMO still the better way is to put this creation into the etcd service unit. This puts it closer to the subject (so it is easier to debug, making the config more self-contained). But I understand the rationale to keep it this way. I'll let other decide here then :+1:

Reply:
I'm fine either way. I think @surajssd approach makes sense, as it is the exact same thing Flatcar is doing and we will remove when it is included in enough Flatcar versions.

I'd rather move to something different when we move away from etcd-wrapper. For now, having the same patch Flatcar has applied seems good, IMHO. But maybe I'm just being lazy :)

Reply:
Container Linux relies on systemd-tmpfiles to create all application folders under `/etc` or `/var` because the root filesystem is created empty by Ignition.

Reply:
@pothos I see. Makes total sense, then. Thanks! =)